src/app/dataset-create/dataset-create.service.ts

```diff
@@ -1,13 +1,15 @@
 import {
```

Comment:
I think we should split this service: dataset create vs dataset commit services.
Currently 2 aspects (create directly or from snapshot vs commit/update) are not cohesive much.

Reply:
Add dataset-commit.service.ts